src/main/java/org/openpnp/gui/LogPanel.java

```diff
@@ -1,175 +1,289 @@
-        }
-    };
+    private boolean autoScroll = PREF_LOG_AUTO_SCROLL_DEF;
+    private boolean systemOutEnabled = true;
+
+    private LogEntryListModel logEntries = new LogEntryListModel();
+    private JList<LogEntry> logEntryJList = new JList<>(logEntries);
+
+    private Level filterLogLevel = Level.TRACE;
+    private LogEntryListModel.LogEntryFilter logLevelFilter = new LogEntryListModel.LogEntryFilter();
+    private LogEntryListModel.LogEntryFilter searchBarFilter = new LogEntryListModel.LogEntryFilter();
+    private LogEntryListModel.LogEntryFilter systemOutFilter = new LogEntryListModel.LogEntryFilter();
 
     public LogPanel() {
+
+        loadLoggingPreferences();
+
+        logEntries.addFilter(logLevelFilter);
+        logEntries.addFilter(searchBarFilter);
+        logEntries.addFilter(systemOutFilter);
+
         setLayout(new BorderLayout(0, 0));
 
-        JToolBar toolBar = new JToolBar();
-        add(toolBar, BorderLayout.NORTH);
+        JPanel settingsAndFilterPanel = new JPanel();
+        settingsAndFilterPanel.setLayout(new BorderLayout(0, 0));
+        add(settingsAndFilterPanel, BorderLayout.NORTH);
+
+        // The Global Filter settings
+        JPanel settingsPanel = new JPanel(new FlowLayout(FlowLayout.LEFT));
+
+        settingsPanel.setBorder(new TitledBorder(new EtchedBorder(EtchedBorder.LOWERED, null, null),
+                "Global Logging Settings", TitledBorder.LEADING, TitledBorder.TOP, null, new Color(0, 0, 0)));
+
+        settingsPanel.add(createGlobalLogLevelPanel());
+
+        settingsAndFilterPanel.add(settingsPanel, BorderLayout.NORTH);
+
+        // The filter settings
+        JPanel filterPanel = new JPanel(new BorderLayout(0, 0));
+
+        filterPanel.setBorder(new TitledBorder(new EtchedBorder(EtchedBorder.LOWERED, null, null),
+                "Filter Logging Panel", TitledBorder.LEADING, TitledBorder.TOP, null, new Color(0, 0, 0)));
+
+        JPanel filterContentPanel = new JPanel(new FlowLayout(FlowLayout.LEFT));
+
+        filterContentPanel.add(createSearchFieldPanel());
+        filterContentPanel.add(createFilterLogLevelPanel());
+
+        filterContentPanel.add(createSystemOutputCheckbox());
+
+        filterPanel.add(filterContentPanel, BorderLayout.WEST);
+
+        JPanel filterControlPanel = new JPanel(new FlowLayout(FlowLayout.RIGHT));
+
+        JButton btnClear = new JButton("Clear Panel");
+        btnClear.addActionListener(e -> logEntries.clear());
+
+        filterControlPanel.add(btnClear);
+
+        JButton btnCopyToClipboard = new JButton("Copy to clipboard");
+        btnCopyToClipboard.addActionListener(e -> {
+            StringBuilder sb = new StringBuilder();
+            logEntries.getFilteredLogEntries().forEach(logEntry -> sb.append(logEntry.getRenderedLogEntry()));
+            copyStringToClipboard(sb.toString());
+        });
+
+        filterControlPanel.add(btnCopyToClipboard);
+
+        JToggleButton btnScroll = new JToggleButton("Auto Scroll");
```

Comment:
Maybe make this a new icon. Just noting it here, if you don't want to mess with icons I can always throw that in later.